addons/s3/models.py

```diff
@@ -135,10 +157,17 @@ def serialize_waterbutler_credentials(self):
         }
 
     def serialize_waterbutler_settings(self):
+        """
+        We use the folder id to hold the bucket location
+        """
+
+        bucket_name = self.folder_id.split('/')[0]
```

Comment:
nit: could be a case for `string.partition`. Something like `bucket_name, _, folder_id' = folder_id.partition('/')`